DitoSDK/sdk/src/main/java/br/com/dito/ditosdk/notification/NotificationOpenedReceiver.kt

```diff
@@ -17,23 +17,29 @@ class NotificationOpenedReceiver: BroadcastReceiver() {
         }
 
         intent?.getStringExtra(Dito.DITO_NOTIFICATION_ID)?.let {
-            Dito.notificationRead(it)
+            notificationId = it
         }
 
+        intent?.getStringExtra(Dito.DITO_NOTIFICATION_REFERENCE)?.let {
+            reference = it
+        }
+
+        if (reference != "" && notificationId != "") {
+            Dito.notificationRead(notificationId, reference)
+        }
 
         context?.let {
             getIntent(it)?.let {
-                context.startActivity(intent)
+                context.startActivity(it)
```

Comment:
@Danielsilveira98 Pq mudamos isso?

Reply:
`getIntent` retorna o intent, e o `.let` faz o `intent` retornado ser jogado para a variável `it` . Da forma como estava antes, não estava encontrando o `intent`. Então alteramos que que o `intent` ativiado seja o que foi retornado pelo `getIntent`.